### DISCUSSION TITLE
Errors when generating images using DALL-E-3

### DISCUSSION THREAD

**Discussion by k-toki** [danny-avila/LibreChat#5027] [Troubleshooting] [answered]
### What happened?

I get the following error message when generating images using DALL-E-3 in Plugin. Please tell me how to deal with this.

Something went wrong. Here's the specific error message we encountered: Encountered an error while attempting to respond: 404 This is not a chat model and thus not supported in the v1/chat/completions endpoint. Did you mean to use v1/completions?

In another environment, another message appears with the following error Please let me know about this as well.

Something went wrong. Here's the specific error message we encountered: Encountered an error while attempting to respond: 400 Unsupported value: ' messages[0].role' does not support 'system' with this model.

Translated with DeepL.com (free version)

### Steps to Reproduce

1.Select PlugIns
2.Enter prompts for image generation
3.An error is displayed.

### What browsers are you seeing the problem on?

Chrome

### Screenshots

![12-18-2024_10_56_AM](https://github.com/user-attachments/assets/254a03a7-b080-484b-bb4b-289cc940ed54)
![12-18-2024_10_55_AM](https://github.com/user-attachments/assets/79bfcc06-a087-4d3e-8fa7-1eb6bff030fc)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Not enough information here. What models are you using? Is this a reverse proxy? Also, look into using Agents, plugins are now deprecated:
https://www.librechat.ai/docs/features/agents

**Reply by k-toki** [reply]
Thank you, I was able to solve the problem using Assistant instead of the plugin.

**Reply by brocodevs** [reply]
could you elaborate? im having the same issue...

**Reply by berry-13** [reply]
To fix this, you'll need to either use Agents or OpenAI's assistants. We highly recommend going with Agents since they’re fully local and have better support compared to endpoints like OpenAI Assistant.

If you’re unsure how to create an Agent, check out the guide Danny shared earlier, or ask about it on Github/Discord by starting a new discussion.

To add the DALL-E-3 plugin, open the Agent Builder, click on "Add Tools," and then hit "Add" under the DALL-E-3 plugin option

**Reply by brocodevs** [reply]
Thanks for the quick response! I'd created an agent prior to this, but id gotten another error. After a bunch of fiddling and switching accounts, it seems to have fixed itself somehow. Thanks!

**Reply by igorcafe** [reply]
I see so plugins are basically agent tools now, but some are missing and plugins seems to be disabled now in master
